bigbite/themer
# [BBT-99] Tidy up front end

## Description

[BBT-99](https://b5ecom.atlassian.net/browse/BBT-99) - There was a request to tidy up the UI for a demo but this tidying up could potentially be added into the RC. Discussion to be had on this PR on these changes.

See change log for a full details of what has changed.

## Change Log

- Changes `getCoreBlocks` in `Blocks.js` to use its first mode. This gets all the blocks and doesn't use the `getCoreBlocksFromSchema` function considering we are now not generating the controls based on the schema anymore this doesn't seem necessary. This also means we can show the nice name and icon for each block.
- Only shows a block for an editor if that block has controls to be displayed.
- Displays Block Name and Block Icon.
- Adds "More Menu" dropdown to house the Export Tool.
- For the preview, I've split the two paragraphs into two paragraph blocks (as they should be) and also added a quote and pull-quote block.
- Adds a chevron icon to the accordion items that turns 90 degrees when open.
- Updates `@wordpress/icons` package.

## Steps to test

- Check blocks are loaded in and only the blocks that have styles (controls) to be changed.
- Check blocks in this list show nice name and logo.
- Check "More Menu" shows next to "Save" button, opens and Export functions correctly (saves new theme.json).
- Check preview has new paragraph, quote and pull-quote blocks and they display correctly.
- Check accordion items have chevron and the chevron turns 90 degrees when that item is selected (opened).

## Screenshots/Videos

### Block List
| **Old**  | **New** |
| - | - |
| <img width="740" alt="Screenshot 2023-10-09 at 14 30 38" src="https://github.com/bigbite/themer/assets/5260744/57ac26b5-d95b-416d-9c61-08452cc7682e">  | <img width="745" alt="Screenshot 2023-10-09 at 14 27 57" src="https://github.com/bigbite/themer/assets/5260744/d56e30f3-50af-46a5-b529-0a8883932d93">  |

### More Menu
| **Old**  | **New** |
| - | - |
| <img width="787" alt="Screenshot 2023-10-09 at 14 33 41" src="https://github.com/bigbite/themer/assets/5260744/accd8b68-29e6-41c8-a4df-9dc0dc24cf8a"> | <img width="285" alt="Screenshot 2023-10-09 at 14 27 48" src="https://github.com/bigbite/themer/assets/5260744/87dcd442-7f98-466b-b923-ba91dac2da82">  |

### Block Preview
| **Old**  | **New** |
| - | - |
| <img width="780" alt="Screenshot 2023-10-09 at 14 33 56" src="https://github.com/bigbite/themer/assets/5260744/03b76702-1ee5-4086-9455-5d844ec90970">  | <img width="784" alt="Screenshot 2023-10-09 at 14 27 39" src="https://github.com/bigbite/themer/assets/5260744/417e67e1-919f-4ba3-8dfb-16979d8e56d6">  |

### Block List Open
| **Old**  | **New** |
| - | - |
| <img width="740" alt="Screenshot 2023-10-09 at 14 34 17" src="https://github.com/bigbite/themer/assets/5260744/5257af23-fc19-4306-8afa-bdb99b7f925d">  | <img width="740" alt="Screenshot 2023-10-09 at 14 28 11" src="https://github.com/bigbite/themer/assets/5260744/7b9ec89c-b7fd-41ca-81ba-7cb5d62b1908">  |

### Video Demo
https://github.com/bigbite/themer/assets/5260744/f00729fe-fd01-4bce-9719-ef1dcd793a13


## Checklist:





[BBT-99]: https://b5ecom.atlassian.net/browse/BBT-99?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
